daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -289,13 +290,16 @@ final class ImmArray[+A] private (
   }
 
   /** O(n) */
-  def find(f: A => Boolean): Option[A] =
-    indices collectFirst (Function unlift { i =>
+  def find(f: A => Boolean): Option[A] = {
+    var i = 0
+    var found: Option[A] = None
+    while (i < len && found.isEmpty) {
```

Comment:
looks not as nice as original way, but much quicker

Reply:
Note a big deal, but I would prefer a tail recursive function, something like that:

```
def find(f: A => Boolean): Option[A] = {
  @tailrec
  go(i :Int = 0): Option[A] = 
    if (i >= len) None
    else if (f(e1)) Some(e1)
    else go(i+1)
  go()
}
```

Reply:
Not a big deal, but I would prefer a tail recursive way. something like : 

```
def find(p: A => Boolean): Option[A] = {
  @tailrec 
  go(i: Int): Option[A] = 
    if (i < len) {
       val e = uncheckedGet(i)
       if (p(e)) Some(e)
       else go(i+1)
    } 
    else None 
  go(0)
```

Reply:
ScalaMeter showed me the following results (tested with ImmArray of 1m entries):
1. The original code
Parameters(number of iterations -> 150000000): 8.776301 ms
2. My modification
Parameters(number of iterations -> 150000000): 3.860562 ms
3. Your proposed solution
Parameters(number of iterations -> 150000000): 3.972929 ms

Your proposed solution is similar-ish performance, on my next benchmark run it was actually quicker than my solution. 
Thank you for suggestion, I'll commit it